core/jobs/batch_jobs/voiceover_migration_job.py

```diff
@@ -0,0 +1,314 @@
+                )
+            )
+        entity_voiceovers_model.update_timestamps()
+        return entity_voiceovers_model
+
+    @classmethod
+    def create_entity_voiceovers_instances(
+        cls,
+        exploration_model: exp_models.ExplorationModel,
+        exploration_voice_artists_link_model: (
+            voiceover_models.ExplorationVoiceArtistsLinkModel),
+        voice_artist_metadata_models_list: List[
+            voiceover_models.VoiceArtistMetadataModel]
+    ) -> List[voiceover_domain.EntityVoiceovers]:
+        """Creates a list of EntityVoiceovers instances containing voiceovers
+        from the given exploration data.
+
+        Args:
+            exploration_model: ExplorationModel. An instance of an
+                exploration model.
+            exploration_voice_artists_link_model:
+                ExplorationVoiceArtistsLinkModel. An instance of an
+                ExplorationVoiceArtistsLinkModel model.
+            voice_artist_metadata_models_list: list(VoiceArtistMetadataModel). A
+                list of VoiceArtistMetadataModel instances. Language accent code
+                for a given voice artist should be fetched from this model.
+
+        Returns:
+            list(EntityVoiceovers). A list of EntityVoiceovers instances
+            containing voiceovers from the given exploration data.
+
+        Raises:
+            Exception. All voice artists are not assigned accents in the
+                VoiceArtistMetadataModel.
+        """
+
+        voice_artist_id_to_language_code_mapping = {}
+
+        for voice_artist_metadata_model in voice_artist_metadata_models_list:
+            voice_artist_id_to_language_code_mapping[
+                voice_artist_metadata_model.id] = (
+                    voice_artist_metadata_model.language_code_to_accent)
+
+        entity_type = 'exploration'
+        entity_id = exploration_model.id
+        entity_version = exploration_model.version
+
+        entity_voiceovers_id_to_entity_voiceovers: Dict[
+            str, voiceover_domain.EntityVoiceovers] = {}
+
+        content_id_to_voiceovers_mapping = (
+            exploration_voice_artists_link_model.
+            content_id_to_voiceovers_mapping)
+
+        for content_id, language_code_to_voiceover_mapping in (
+                content_id_to_voiceovers_mapping.items()):
+            for language_code, voiceover_mapping in (
+                    language_code_to_voiceover_mapping.items()):
+
+                voice_artist_id = voiceover_mapping[0]
+                voiceover_dict = voiceover_mapping[1]
+                manual_voiceover_instance = state_domain.Voiceover.from_dict(
+                    voiceover_dict)
+
+                try:
+                    language_accent_code = (
+                        voice_artist_id_to_language_code_mapping[
+                            voice_artist_id][language_code]
+                    )
+                except Exception as e:
```

Comment:
Should this be catching KeyError, since that is what your error message seems to be guarding against?